### DISCUSSION TITLE
Implement KYC Option

### DISCUSSION THREAD

**Discussion by CRVNE** [snapshot-labs/snapshot#1004] [Ideas]
Even if it was a paid service the ability to require KYC for voting or proposal submission would be extremely useful. Especially considering the state of the blockchain space in regards to DAOs & legal voting requirements in regards to securities.

Snapshot could be a 3rd party KYC provider and charge a subscription fee to organizations that require it which could be used to further fund & support Snapshot as a whole. My organization would absolutely pay for the service. I know many others would as well.

**Comment by heyjonmanning** [comment]
This would be amazing. Either KYC offered by Snapshot or being able to connect a 3rd party KYC service that's integrated into the Snapshot platform. I feel like it's the only major feature that Snapshot is missing currently but it's a really important piece for DAOs operating in the U.S. and likely other places as well.

**Comment by bonustrack** [comment]
@cranetechnet @heyjonmanning What is your use case? Are you willing to do voting with your community with only KYC users? What are your communities? This can work with a voting strategies but it require a partner to do the KYC verification and link it to wallet addresses.

**Reply by CRVNE** [reply]
We have a legally established DAO and one of the legal requirements for our country is that in order to vote or claim rewards KYC must be completed first. 

I figured it would be done through a 3rd party, but basically what would be required is the option to complete KYC with ownership of a token. If KYC is completed AND you hold one of the tokens then you would be allowed to vote. If you transfer the token to new wallet then KYC would need to be completed again under the new wallet.

**Comment by jasperverbeet** [comment]
KYC would be very helpful for preventing spam/unwanted proposals. Other options would be some sort of moderation. Currently, we're extracting positive and negative sentiment from the choices using complex regexes (e.g., "Agree" or "👍" would be positive), and if most people vote negatively we might hide the proposal. But that seems like a work around... 

TLDR; A good moderating system (e.g., report spam, upvote/downvote) or KYC in general would be very helpful with preventing spam.
